hack/generator/pkg/astmodel/file_definition.go

```diff
@@ -40,18 +40,19 @@ func NewFileDefinition(packageRef *PackageReference, definitions ...TypeDefiner)
 // generateImports products the definitive set of imports for use in this file and
 // disambiguates any conflicts
 func (file *FileDefinition) generateImports() map[PackageImport]struct{} {
-
-	metav1Import := NewPackageImport(
-		*NewPackageReference("k8s.io/apimachinery/pkg/apis/meta/v1")).WithName("metav1")
-
 	var requiredImports = make(map[PackageImport]struct{}) // fake set type
-	requiredImports[*metav1Import] = struct{}{}
 
 	for _, s := range file.definitions {
-		for _, requiredImport := range s.Type().RequiredImports() {
+		for _, requiredImport := range s.RequiredImports() {
 			// no need to import the current package
 			if !requiredImport.Equals(file.packageReference) {
 				newImport := NewPackageImport(*requiredImport)
+
+				// TODO: Should we change signature of RequiredImports to return this type instead?
```

Comment:
I'm wondering if we should change RequiredImports to return the `PackageImport` type rather than the `PackageReference` type - that way the type itself can specify what it wants the import name to be, rather than this hack we have here where the renaming happens above.

If there were multiple imports that were the same but with different names we can either error on that, or just pick one and let the types figure it out using `codeGenerationContext` (which they should be able to since the context will have the winner)

Reply:
If folks like this plan/idea I'll do it in this PR too

Reply:
Actually on second thought maybe this doesn't make sense.

definitions/types identify what packages they require. `CodeGenerationContext` is built from that set, naming and disambiguating as required. The `AsAst` methods then all use `CodeGenerationContext` to pick out the name of the package chosen - so there shouldn't be any expectation that a particular package got mapped to a particular name, should always just defer to `CodeGenerationContext` probably, which means what we're doing now is right.